{
  "schema_version": "1.5.0",
  "id": "CVE-2021-3177",
  "modified": "2023-05-24T00:00:00",
  "published": "2021-01-19T00:00:00",
  "summary": "ctypes: Buffer overflow in PyCArg_repr",
  "details": "Python 3.x through 3.9.1 has a buffer overflow in PyCArg_repr in _ctypes/callproc.c, which may lead to remote code execution in certain Python applications that accept floating-point numbers as untrusted input, as demonstrated by a 1e300 argument to c_double.from_param. This occurs because sprintf is used unsafely.",
  "affected": [
    {
      "ranges": [
        {
          "type": "GIT",
          "events": [
            {
              "introduced": "0"
            },
            {
              "fixed": "34df10a9a16b38d54421eeeaf73ec89828563be7"
            },
            {
              "fixed": "916610ef90a0d0761f08747f7b0905541f0977c7"
            },
            {
              "fixed": "c347cbe694743cee120457aa6626712f7799a932"
            },
            {
              "fixed": "d9b8f138b7df3b455b54653ca59f491b4840d6fa"
            },
            {
              "fixed": "ece5dfd403dac211f8d3c72701fe7ba7b7aa5b5f"
            }
          ],
          "repo": "https://github.com/python/cpython"
        }
      ]
    }
  ],
  "references": [],
  "database_specific": {
    "cwe_ids": []
  }
}